front/app/components/InitiativeCards/SortFilterDropdown.tsx

```diff
@@ -1,67 +1,79 @@
-import React, { PureComponent } from 'react';
+import React, { useState, useEffect } from 'react';
 
 import FilterSelector from 'components/FilterSelector';
 
-import { FormattedMessage } from 'utils/cl-intl';
+import { ScreenReaderOnly } from 'utils/a11y';
+import { FormattedMessage, MessageDescriptor, useIntl } from 'utils/cl-intl';
 
 import messages from './messages';
 
 export type Sort = 'random' | 'likes_count' | 'new' | '-new';
 
+const optionMessages: { [key in Sort]: MessageDescriptor } = {
+  random: messages.random,
+  likes_count: messages.popular,
+  new: messages.newest,
+  '-new': messages.oldest,
+};
+
 type Props = {
-  id?: string | undefined;
   alignment: 'left' | 'right';
   defaultSortingMethod?: Sort;
   onChange: (value: Sort) => void;
 };
 
-type State = {
-  selectedValue: string[];
-};
-
-class SortFilterDropdown extends PureComponent<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      selectedValue: [props.defaultSortingMethod ?? 'new'],
-    };
-  }
+const SortFilterDropdown = ({
+  alignment,
+  defaultSortingMethod = 'new',
+  onChange,
+}: Props) => {
+  const [selectedValue, setSelectedValue] = useState<Sort[]>([
+    defaultSortingMethod,
+  ]);
+  const { formatMessage } = useIntl();
 
-  handleOnChange = (selectedValue: Sort[]) => {
-    this.setState({ selectedValue });
-    this.props.onChange(selectedValue[0]);
+  const handleOnChange = (value: Sort[]) => {
+    setSelectedValue(value);
+    onChange(value[0]);
   };
 
-  render() {
-    const { alignment } = this.props;
-    const { selectedValue } = this.state;
-    const options = [
-      { text: <FormattedMessage {...messages.random} />, value: 'random' },
-      {
-        text: <FormattedMessage {...messages.popular} />,
-        value: 'likes_count',
-      },
-      { text: <FormattedMessage {...messages.newest} />, value: 'new' },
-      { text: <FormattedMessage {...messages.oldest} />, value: '-new' },
-    ];
+  const options = Object.keys(optionMessages).map((key) => {
+    return {
+      text: <FormattedMessage {...optionMessages[key]} />,
+      value: key,
+    };
+  });
+
+  useEffect(() => {
+    setSelectedValue([defaultSortingMethod]);
+  }, [defaultSortingMethod]);
 
-    return (
+  return (
+    <>
+      <ScreenReaderOnly aria-live="assertive">
+        <FormattedMessage
+          {...messages.a11y_itemsHaveChanged}
+          values={{
+            sortOder: formatMessage(optionMessages[selectedValue[0]]),
```

Comment:
```suggestion
            sortOrder: formatMessage(optionMessages[selectedValue[0]]),
```
Also in other spots. But I guess translations are already pending. Then you would have to trigger new translations. So probably never mind.